spring-rabbit/src/main/java/org/springframework/amqp/rabbit/connection/CachingConnectionFactory.java

```diff
@@ -1295,4 +1316,23 @@ else if (!RabbitUtils.isNormalChannelClose(cause)) {
 
 	}
 
+
+    /**
+     * Name each create thread by this class.
+     */
+    private static class AmqpNamedThreadPoolFactory implements ThreadFactory {
+        private AtomicInteger threadId = new AtomicInteger();
+
+        @Override
+        public Thread newThread(Runnable r) {
+            Thread t = new Thread(r);
+            t.setDaemon(true);
+            t.setName("spring-rabbit-deferred-pool-"
+                    + threadPoolId.incrementAndGet()
+                    + "-thread-"
```

Comment:
We should include the connection factory beanName (if present) in the thread names. Will need a protected `getBeanName()` on the abstract factory and pass it into this class.

Reply:
I think we don't need `stsatic threadPoolId` counter if we are going to use a `beanName` as base.

Reply:
(Except when it's not a bean 😄 )

Reply:
Then call `setBeanName()` manually! 😃